app/src/main/java/ar/com/wolox/android/example/ui/home/news/NewsFragment.kt

```diff
@@ -10,6 +14,27 @@ class NewsFragment @Inject constructor() : WolmoFragment<FragmentNewsBinding, Ne
     override fun layout() = R.layout.fragment_news
 
     override fun init() {
+        with(binding) {
+            recyclerView.layoutManager = LinearLayoutManager(context)
+            val newsAdapter = NewsAdapter(newsList(resources))
+            recyclerView.adapter = newsAdapter
+            swipeRefresh.setColorSchemeResources(R.color.colorAccent)
+            if (newsAdapter.itemCount > 0) {
+                recyclerView.visibility = View.VISIBLE
+                noNewsMessage.visibility = View.GONE
+            } else {
+                recyclerView.visibility = View.GONE
+                noNewsMessage.visibility = View.VISIBLE
+            }
+        }
+    }
+
+    override fun setListeners() {
+        with(binding) {
+            swipeRefresh.setOnRefreshListener(OnRefreshListener {
+                swipeRefresh.isRefreshing = false
```

Comment:
This should be a decission made in the presenter. The fragment just has the responsibility to hide and show the refreshing icon